app/src/main/java/com/example/ubclaunchpad/launchnote/photoBrowser/PhotoBrowserActivity.kt

```diff
@@ -87,21 +88,17 @@ class PhotoBrowserActivity : BaseActivity(), AllPhotosFragment.OnEditPhotoMode,
                 }
     }
 
-    override fun onEditPhotoMode(isActiveEdit: Boolean, selectedImages: Set<PicNote>) {
-        Observable.just(isActiveEdit)
-                .subscribeOn(AndroidSchedulers.mainThread())
-                .subscribe {
-                    if (it) {
-                        toolbarFragment.setMode(PhotoNavigatonToolbarFragment.ToolbarMode.EditMode)
-                    } else {
-                        toolbarFragment.setMode(PhotoNavigatonToolbarFragment.ToolbarMode.NormalMode)
-                    }
-                }
+    override fun onEditPhotoMode(isActiveEdit: Boolean, imagesSelected: Set<PicNote>) {
+        if (isActiveEdit) {
```

Comment:
Either modify the caller to call this fn on the UI thread or keep it the way it was otherwise it will blow up since setting the mode of the toolbar needs to run on the UI thread.

Try opening the edit mode and then click on delete and it will throw a runtime exception